### PR TITLE
Add deaths timeseries for San Mateo

### LINKED ISSUE #171: The San Mateo deaths data v2 is missing
**Describe the bug**
The deaths data for San Mateo County in data v2 is missing:
https://github.com/sfbrigade/stop-covid19-sfbayarea/blob/master/data/data.v2.json#L16204

**To Reproduce**
1. Go to data.v2.json file
2. Search for "san_mateo" and check the "deaths" array
3. The deaths array has no elements

**Expected behavior**
To move from data v1 to data v2, we need the cases and deaths information for each county

**Screenshots**
Screen Shot 2021-01-06 at 9 27 55 PM

### PR DESCRIPTION
**Pull request by Mr0grog**
San Mateo doesn't publish a timeseries for deaths, so we now source that information from the state's data portal. As part of doing so, we've added a lightweight abstraction for querying CKAN, which simplifies some of the hospital data code (since it also uses the state's data portal).

Fixes #171.

### CODE REVIEW

**Review by benghancock — APPROVED** [review]
This looks good @Mr0grog! The wrapper around the CKAN API is nice. I added just one question inline above. Thanks for integrating that into the hospital scraper, too!

One thing I noticed about this data is that the state appears to sometimes log negative `newcountdeaths` values (see for example on 2020-03-30), which then impact the `totalcountdeaths`. At first I thought it was a side effect of us converting null values to `-1` but I double-checked the state website and those values are real. Not sure what that's meant to reflect but possibly correcting prior figures?

Anyway, good to go! :+1:

File: covid19_sfbayarea/data/ckan.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            types) as the first item.
+        params
+            Any other data search arguments. For a full reference, see:
+            https://docs.ckan.org/en/2.7/maintaining/datastore.html#ckanext.datastore.logic.action.datastore_search
+        """
+        params['resource_id'] = resource_id
+        if isinstance(params.get('q'), dict):
+            params['filters'] = json.dumps(params['filters'])
+        if isinstance(params.get('filters'), dict):
+            params['filters'] = json.dumps(params['filters'])
```
[the remark this answers is not in the document]
Comment (Mr0grog) [reply covid19_sfbayarea/data/ckan.py:42]: Augh! That first one is totally wrong! It should be:

```suggestion
        if isinstance(params.get('q'), dict):
            params['q'] = json.dumps(params['q'])
        if isinstance(params.get('filters'), dict):
            params['filters'] = json.dumps(params['filters'])
```

**Comment by Mr0grog** [comment]
> the state appears to sometimes log negative `newcountdeaths` values (see for example on 2020-03-30), which then impact the `totalcountdeaths`. At first I thought it was a side effect of us converting null values to `-1` but I double-checked the state website and those values are real. Not sure what that's meant to reflect but possibly correcting prior figures?

Whaaaaaaaat. Wow, I somehow did not notice that was happening at all. The total on the current day matches the county’s total, so yeah… I guess this isn’t a mistake. I would guess the same thing as you.

**Comment by Mr0grog** [comment]
Rebased on master to make this mergeable.

**Pull request merged**